src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -48,4 +51,21 @@ public String getBucket() {
     return bucket;
   }
 
+  @Override
+  public void validate() {
+    super.validate();
+    if (!containsMacro("bucket") && bucket != null) {
+      String bucketName = BigQueryUtil.getBucketNameWithoutScheme(bucket);
+      if (bucketName.isEmpty()) {
+        throw new InvalidConfigPropertyException("Bucket name must be of format gs://<bucket> or <bucket>.",
+                                                 "bucket");
+      }
+      // Basic validation for allowed characters as per https://cloud.google.com/storage/docs/naming
+      Pattern p = Pattern.compile("[a-z0-9._-]+");
+      if (!p.matcher(bucket).matches()) {
+        throw new InvalidConfigPropertyException("Bucket name must be a valid bucket name of format " +
+                                                   "gs://<bucket> or <bucket>", "bucket");
```

Comment:
This happens if the bucket contains unsupported characters. A better error message would be something like:
"Bucket names can only contain lowercase characters, numbers, '.', '_', and '-'."

Also, is there a length limit for buckets?

Reply:
This check is just a basic allowed character check, which is just a subset of all the constraints applied to gcs bucket name. This is done to fail fast if there are any characters in the bucket name which are not allowed.
https://cloud.google.com/storage/docs/naming